Modules/Segmentation/MarkovRandomFieldsClassifiers/include/itkRGBGibbsPriorFilter.hxx

```diff
@@ -623,11 +615,8 @@ RGBGibbsPriorFilter<TInputImage, TClassifiedImage>::RegionEraser()
 {
   const unsigned int size = m_ImageWidth * m_ImageHeight * m_ImageDepth;
 
-  delete[] m_Region;
-  m_Region = new unsigned short[size];
-
-  delete[] m_RegionCount;
-  m_RegionCount = new unsigned short[size];
```

Comment:
These original lines of code don't look exception-safe, because if either of the two `new` statements would fail (by throwing an `std::bad_alloc` exception), one of the raw pointers `m_Region` or `m_RegionCount` would remain "dangling", i.e., pointing to deallocated memory. The original `~RGBGibbsPriorFilter()` would then try to deallocated memory that was already deallocated, causing undefined behavior (typically crashing).